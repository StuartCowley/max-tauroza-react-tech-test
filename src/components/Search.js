import React, { useState } from 'react';
import '../styles/Search.css';

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState('');

  const handleInputChange = (e) => setValue(e.target.value);

  return (
    <>
      <form>
        <input
          className="searchBar"
          type="text"
          placeholder="Search"
          onChange={handleInputChange}
        />
        <button className="searchButton" type="submit">
          Submit
        </button>
      </form>
    </>
  );
};

export default Search;
