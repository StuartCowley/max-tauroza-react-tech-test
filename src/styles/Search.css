form {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
}

.searchBar {
  align-self: center;
  margin-left: auto;
  margin-top: 20px;
  margin-bottom: 20px;
  padding: 0 10px;
  font-size: 20px;
  border-radius: 30px;
  border-style: solid;
  border-bottom-width: 3px;
  border-top-width: 1px;
  border-left-width: 1px;
  border-right-width: 1px;
  border-color: midnightblue;
  font-family: 'Audiowide';
}

.searchButton {
  margin-right: auto;
  margin-left: 10px;
  margin-top: 20px;
  margin-bottom: 20px;
  padding: 0 10px;
  font-size: 20px;
  border-radius: 30px;
  align-self: center;
  font-family: 'Audiowide';
}

@media only screen and (max-width: 615px) {
  form {
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  .searchBar {
    margin: auto;
    width: 30vw;
    font-size: 3vw;
  }
  .searchButton {
    margin: auto;
    width: 12vw;
    font-size: 2vw;
  }
}
