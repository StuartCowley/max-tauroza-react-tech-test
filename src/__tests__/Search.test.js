import { render, screen } from '@testing-library/react';
import Search from '../components/Search';

describe('Search', () => {
  it('renders app component correctly', () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a search input bar', () => {
    render(<Search />);
    expect(screen.getByPlaceholderText('Search')).toBeInTheDocument();
  });
});
