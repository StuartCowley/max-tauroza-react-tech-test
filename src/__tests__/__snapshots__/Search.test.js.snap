// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders app component correctly 1`] = `
<DocumentFragment>
  <form>
    <input
      class="searchBar"
      placeholder="Search"
      type="text"
    />
    <button
      class="searchButton"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
